lib/post.ml

```diff
@@ -163,7 +163,7 @@ let post_of_atom ~(feed : Feed.t) (e : Syndic.Atom.entry) =
         (List.find (fun l -> l.Syndic.Atom.rel = Syndic.Atom.Alternate) e.links)
           .href
     with Not_found -> (
-      match e.links with l :: _ -> Some l.href | [] -> None)
+      match e.links with l :: _ -> Some l.href | [] -> Some e.id)
```

Comment:
Should we check that id is a valid URL?

Reply:
https://www.ietf.org/rfc/rfc4287.txt

It seems that, even if the URI is valid, there's no guarantee that a resource can be fetched via HTTP (i.e. it's allowed for Atom feeds to declare resources that are not dereferencable).

```
4.2.6.  The "atom:id" Element

   The "atom:id" element conveys a permanent, universally unique
   identifier for an entry or feed.

   atomId = element atom:id {
      atomCommonAttributes,
      (atomUri)
   }

   Its content MUST be an IRI, as defined by [RFC3987].  Note that the
   definition of "IRI" excludes relative references.  Though the IRI
   might use a dereferencable scheme, Atom Processors MUST NOT assume it
   can be dereferenced.
```

Reply:
I added a check on the URI scheme, so that we only recognise this as a link if the scheme is "http" or "https".